source/Condition.h

```diff
@@ -0,0 +1,358 @@
+/* Condition.h
+Copyright (c) 2022 by an anonymous author.
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONDITION_H_
+#define CONDITION_H_
+
+
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <string>
+#include <type_traits>
+
+#include "ConditionsStore.h"
+
+
+
+// A Condition stores either:
+//   1. A condition that is in a ConditionsStore
+//   2. A literal value (key is empty)
```

Comment:
We already support [Value Expressions](https://github.com/endless-sky/endless-sky/wiki/Player-Conditions#value-expressions) intermixed with conditions in other parts of the code, so I expect that we don't need this Class if we re-use a bit more of the code below the value-expressions.

Reply:
Allowing ValueExpressions is a great idea, and I would like to implement it here, but *iteratively*. This PR is too large already, and I want to wait until later PRs to expand the functionality.

Reply:
Consider how you would put a value expression in these three places:

```
fleet
    cargo "&cargo count"
    variant "&variant 1 weight"
       Archon
    variant "&variant 2 weight"
       Sparrow
```

... and then consider how much coding is involved. Such a thing belongs in a later PR.

Reply:
As for replacing all Condition objects with ConditionSets, that would be extremely slow. The Condition class is necessary for efficiency reasons.

Reply:
The FAQ now has a direct response to this, "We don't need Condition. We should use ConditionSet for everything instead." It already dealt with that matter in other questions, but I've added a FAQ entry just for this one now.